src/caffe/test/test_data/generate_sample_data.py

```diff
@@ -36,23 +36,25 @@
     f['label'] = label
     f['label2'] = label2
 
-with h5py.File(script_dir + '/sample_data_2_gzip.h5', 'w') as f:
+with h5py.File(script_dir + '/sample_data_uint8_gzip.h5', 'w') as f:
     f.create_dataset(
         'data', data=data + total_size,
         compression='gzip', compression_opts=1
     )
     f.create_dataset(
         'label', data=label,
-        compression='gzip', compression_opts=1
+        compression='gzip', compression_opts=1,
+        dtype='uint8',
     )
     f.create_dataset(
         'label2', data=label2,
-        compression='gzip', compression_opts=1
+        compression='gzip', compression_opts=1,
+        dtype='uint8',
     )
 
 with open(script_dir + '/sample_data_list.txt', 'w') as f:
-    f.write(script_dir + '/sample_data.h5\n')
-    f.write(script_dir + '/sample_data_2_gzip.h5\n')
+    f.write('src/caffe/test/test_data/sample_data.h5\n')
+    f.write('src/caffe/test/test_data/sample_uint8_gzip.h5\n')
```

Comment:
Oh right, this lets you run the `generate_sample_data.py` script from the `src/caffe/test/test_data/` directory and still generate the correct paths in the text file. Otherwise, if you run the script again, the paths in the textfiles change from:

    src/caffe/test/test_data/sample_data.h5

to:

    /home/lyeager/caffe/caffe/src/caffe/test/test_data/sample_data.h5

I chose to fix it this way. The other way to fix it would be to remove the `os.path.abspath()` call on line 8. Would you rather me fix it that way?